test/spec/Feature/Auth/AudienceJwtSecretSpec.hs

```diff
@@ -3,43 +3,242 @@ module Feature.Auth.AudienceJwtSecretSpec where
+        `shouldRespondWith` 200
+
+  context "when the audience claim is not present" $ do
+    it "succeeds with a JWT with no audience claim" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe"
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+    it "succeeds without a JWT" $
+      get "/has_count_column" `shouldRespondWith` 200
+
+disabledSpec :: SpecWith ((), Application)
+disabledSpec = describe "test handling of aud claims in JWT when the jwt-aud is not set" $ do
+
+  context "when the audience claim is a string" $ do
+    it "ignores the audience claim and suceeds" $ do
+      let jwtPayload =
+            [json|{
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": "youraudience"
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+    it "ignores the audience claim and suceeds when it's empty" $ do
+      let jwtPayload =
+            [json|{
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ""
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+  context "when the audience is an array of strings" $ do
+    it "ignores the audience claim and suceeds when it has 1 element" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["youraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+    it "ignores the audience claim and suceeds when it has more than 1 element" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["notyouraudience", "youraudience", "anotheraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+    it "ignores the audience claim and suceeds when it's empty" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": []
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
```

Comment:
Funnily enough, this test fails for v12 (`jwt-aud` not set and an empty array `aud`). Once the array had elements, it worked. (Just to mention, this should behave the same as discussed above).